examples/vision/deeplabv3_plus.py

```diff
@@ -0,0 +1,298 @@
+"""
+Title: Multiclass Semantic Segmentation using DeepLabV3+
+Author: [Soumik Rakshit](http://github.com/soumik12345)
+Date created: 2021/08/31
+Last modified: 2021/09/1
+Description: Implement DeepLabV3+ architecture for Multi-class Semantic Segmentation
+"""
+"""
+## Introduction
+"""
+
+"""
+Semantic segmentation with the goal to assign semantic labels to every pixel in an image
+is one of the fundamental topics in computer vision. Deep convolutional neural networks
+based on the Fully Convolutional Neural Network show striking improvement over systems
+relying on hand-crafted features on benchmark tasks. In this example, we would implement
+the **DeepLabV3+** model as proposed by the paper [Encoder-Decoder with Atrous Separable
+Convolution for Semantic Image Segmentation](https://arxiv.org/pdf/1802.02611.pdf) for
+multi-class semantic segmentation.
+"""
+
+"""
+## Download Dataset
+"""
+
+import os
+import cv2
+import numpy as np
+from glob import glob
+import tensorflow as tf
+from scipy.io import loadmat
+import matplotlib.pyplot as plt
+
+"""shell
+!gdown https://drive.google.com/uc?id=1B9A9UCJYMwTL4oBEo4RZfbMZMaZhKJaz
+!unzip -q instance-level-human-parsing.zip
+"""
+
+"""
+
+"""
+
+"""
+## Building Tensorflow Dataset
+"""
+
+"""
+For this example, we would be using 200 images from the instance-level human parsing
+dataset to train our model.
+"""
+
+IMAGE_SIZE = 512
+BATCH_SIZE = 4
+NUM_CLASSES = 20
+DATA_DIR = "./instance-level_human_parsing/instance-level_human_parsing/Training"
+MAX_IMAGES = 200
+
+train_images = sorted(glob(os.path.join(DATA_DIR, "Images/*")))[:MAX_IMAGES]
+train_masks = sorted(glob(os.path.join(DATA_DIR, "Category_ids/*")))[:MAX_IMAGES]
+
+
+def read_image(image_path, mask=False):
+    image = tf.io.read_file(image_path)
+    if mask:
+        image = tf.image.decode_png(image, channels=1)
+        image.set_shape([None, None, 1])
+        image = tf.image.resize(images=image, size=[IMAGE_SIZE, IMAGE_SIZE])
+        image = tf.cast(image, tf.float32)
```

Comment:
Is the cast necessary? The output of `resize` should be float32

Reply:
Remove unnecessary cast.